arch/risc-v/src/litex/litex_irq_dispatch.c

```diff
@@ -114,13 +114,13 @@ void *litex_dispatch_irq(uint32_t vector, uint32_t *regs)
 #endif
 
   /* If a context switch occurred while processing the interrupt then
-   * g_current_regs may have change value.  If we return any value different
+   * CURRENT_REGS may have change value.  If we return any value different
    * from the input regs, then the lower level will know that a context
    * switch occurred during interrupt processing.
    */
 
-  regs = (uint32_t *)g_current_regs;
-  g_current_regs = NULL;
+  regs = (uintptr_t *)CURRENT_REGS;
```

Comment:
I think cast is not needed

```suggestion
  regs = CURRENT_REGS;
```

Reply:
We should keep this cast to avoid warning "assignment discards 'volatile' qualifier from pointer target type [-Wdiscarded-qualifiers]".